internal/pkg/vcsutil/vcs_util_test.go

```diff
@@ -0,0 +1,362 @@
+package vcsutil
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/stretchr/testify/require"
+)
+
+type VCSTester struct {
+	repoPath       string
+	testFiles      []*os.File
+	remoteRepoPath string
+	remoteUrl      string
+	remoteName     string
+}
+
+var testHasGit bool
+
+func init() {
+	if _, err := exec.LookPath("git"); err == nil {
+		testHasGit = true
+	}
+}
+
+func generateGitState(branchName string, t *testing.T) (VCSTester, error) {
```

Comment:
I recommend just generating this yourself and loading them up. See `internal/config/funcs` for how we do it for Git there. The reason I recommend that is because its a LOT easier to create more test cases cause you just run normal `git` commands and just basically freeze the state (the `.git` directory) at a certain point, rather than rebuilding it all at runtime.

I'd take the helpers from `internal/config/funcs` that translate `DOTgit` into `.git` (so you can commit to this repo)